### DISCUSSION TITLE
Rider Now Free for Non-Commercial Use

### DISCUSSION THREAD

**Discussion by QuintillusCFC** [C7-Game/Prototype#453] [General]
See the blog post at https://blog.jetbrains.com/blog/2024/10/24/webstorm-and-rider-are-now-free-for-non-commercial-use/

As you probably know, I'm a big advocate of Rider for C# and Godot development, but AFAIK only @Kright and I ponied up for the licenses before.  Technically we could have looked into the previous open-source licenses but for a sporadically active project, it wasn't necessarily worth the effort.

Now though?  There's no reason not to jump on board.  There's a "[How to Set Up Rider](https://github.com/C7-Game/Prototype/wiki/Developing-and-Setting-Up-IDEs#jetbrains-rider)" section in our wiki, and once you've used it for a while, you'll never want to go back to VSCode.

One note is that JetBrains recently enabled a new user interface for Rider and its other products, which more closely resembles VSCode.  If you are migrating from VSCode, you may like it, but IMO the Classic UI is far superior for professional work, a bit busier but better at making its functionality discoverable.  You can install it from their plugins page at https://plugins.jetbrains.com/plugin/24468-classic-ui, or via the plugins search within Rider.

**Comment by WildWeazel** [comment]
I beat you to the punch by minutes on discord! This is great news, and I also hear it recently has GDScript support. After Carthage we're probably due for an updated VM image with Rider, Godot 4.3, and .NET 8 LTS.

**Reply by QuintillusCFC** [reply]
Interesting, I had not heard about the GDScript support.  The blog post on it is at https://www.jetbrains.com/help/rider/Godot.html#working-with-code

Do you know if people are using the VM?  I've just stuck with bare-metal since it's faster and uses less battery if I'm out.  Though it would be somewhat higher initial setup time.

Godot 4.3 already!  Thankfully those minor dot-version upgrades have historically been seamless.

**Reply by WildWeazel** [reply]
Did I say 4.3? I meant 4.4 ;)

I know a handful of people have grabbed it, but I'm not sure if anyone is actively using it. If people will be using it, it might be worth setting it up with Vagrant rather than manually recreating and uploading an image.
